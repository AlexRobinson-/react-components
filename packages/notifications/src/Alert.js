import styled from 'styled-components';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import CalloutStyles from '@zendesk/garden-css-callouts';
import { retrieveTheme } from '@zendesk/garden-react-theming';

import Well from './Well';

const VALIDATION = {
  SUCCESS: 'success',
  WARNING: 'warning',
  ERROR: 'error'
};

/**
 * Supports all `<div>` props
 */
const Alert = styled(Well).attrs({
  className: props =>
    classNames({
      // Validation types
      [CalloutStyles['c-callout--success']]: props.type === VALIDATION.SUCCESS,
      [CalloutStyles['c-callout--warning']]: props.type === VALIDATION.WARNING,
      [CalloutStyles['c-callout--error']]: props.type === VALIDATION.ERROR
    })
})`
  ${props => retrieveTheme('notifications.alert', props)};
`;

Alert.propTypes = {
  recessed: PropTypes.bool,
  floating: PropTypes.bool,
  type: PropTypes.oneOf([VALIDATION.SUCCESS, VALIDATION.WARNING, VALIDATION.ERROR]).isRequired
};

/** @component */
export default Alert;
