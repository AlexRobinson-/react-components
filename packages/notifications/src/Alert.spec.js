import React from 'react';
import { shallow } from 'enzyme';
import Alert from './Alert';

it('should render alert styling', () => {
  const wrapper = shallow(<Alert />)
  expect(wrapper).toMatchSnapshot();
});

describe('validation', () => {
  it('should render success styling correctly', () => {

  });

  it('should render warning styling correctly', () => {

  });

  it('should render error styling correctly', () => {

  });
});
