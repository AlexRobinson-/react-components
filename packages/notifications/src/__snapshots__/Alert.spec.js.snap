// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert validation should render error styling correctly 1`] = `
<Well
  className="Alert-hSMbxE c-callout--error daVApm"
  type="error"
/>
`;

exports[`Alert validation should render success styling correctly 1`] = `
<Well
  className="Alert-hSMbxE c-callout--success daVApm"
  type="success"
/>
`;

exports[`Alert validation should render warning styling correctly 1`] = `
<Well
  className="Alert-hSMbxE c-callout--warning daVApm"
  type="warning"
/>
`;
